lifting-state-up/passing-down-event-handlers.md

```diff
@@ -0,0 +1,250 @@
+
+This code:
+
+- Creates a new piece of state named `studentData`
+- Creates a new update function named `setStudentData`
+- Sets the initial value of `studentData`
+- Passes the value of `studentData` to `StudentList` in the `prop` `students
+
+## Create a Function to Update `studentData` in `App`
+
+We can create as many functions and helper functions inside our React components as we want!
+
+In `App`, let's create a function named `updateStudentData`. It's responsible for taking in the updated data for one student, and updating `studentData` on state.
+
+Since this function updates _one_ student, it needs to receive the updated student data. This function should accept a parameter, `updatedStudentData`.
+
+This function should go inside the `App` component, after `studentData` is defined, and before the `return` statement.
+
+<!-- prettier-ignore-start -->
+```js
+  const updateStudentData = updatedStudent => {
+    const students = studentData.map(student => {
+      if (student.id === updatedStudent.id) {
+        return updatedStudent;
+      } else {
+        return student;
+      }
+    });
+
+    setStudentData(students);
+  }
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:300px;"> Piece of Code </div> | Notes                                                                                                                                                                |
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `const updateStudentData = `                        | We're creating a function named `updateStudentData`.                                                                                                                 |
+| `updatedStudent`                                    | This function accepts one argument: an object that holds updated student data. We expect this object to have the key `id`, at the very least.                        |
+| `const students = ;`                                | For this function, we'll create a helper array, `students` to contain the updated student data.                                                                      |
+| `studentData.map(...)`                              | We'll map over `studentData`...                                                                                                                                      |
+| `student => {...}`                                 | Inside `map`, we'll reference each item with `student`...                                                                                                            |
+| `if (student.id === updatedStudent.id) {`           | If we found the student that our `updatedStudent` is updating...                                                                                                     |
```

Comment:
voice in present tense. since the example has already been shown (past would be ok, but present makes it sound more immediate)
```suggestion
| `const students = ;`                                | For this function, we create a helper array, `students` to contain the updated student data.                                                                      |
| `studentData.map(...)`                              | We map over `studentData`...                                                                                                                                      |
| `student => {...}`                                 | Inside `map`, we reference each item with `student`...                                                                                                            |
| `if (student.id === updatedStudent.id) {`           | If we find the student that our `updatedStudent` is updating...                                                                                                     |
```